apps/src/tutorialExplorer/filterGroupHeaderSelection.jsx

```diff
@@ -0,0 +1,100 @@
+/* filterGroupHeaderSelection: A horizontal selection of filters for a given
+ * filter group, shown in the header on desktop widths.
+ */
+
+import React, {PropTypes} from 'react';
+
+const styles = {
+  container: {
+    display: 'inline-block',
+    marginTop: 6,
+    overflow: 'hidden',
+    height: 34,
+    lineHeight: '34px',
+    border: "solid 1px #cecece",
+    borderRadius: 5
+  },
+  flexContainer: {
+    display: 'flex',
+    flexWrap: 'nowrap'
+  },
+  item: {
+    backgroundColor: "white",
+    color: "dimgrey",
+    fontFamily: "'Gotham 4r', sans-serif",
+    fontSize: 15,
+    cursor: 'pointer',
+    float: 'left',
+    textAlign: "center",
+    flex: 1,
+    userSelect: 'none',
+    boxSizing: 'border-box',
+    borderLeft: "solid 1px white"
+  },
+  select: {
+    backgroundColor: '#2799a4',
+    color: 'white',
+    borderLeft: "solid 1px #2799a4"
+  },
+  borderOnLeft: {
+    borderLeft: "solid 1px #cecece"
+  }
+};
+
+export default class FilterGroupHeaderSelection extends React.Component {
+  static propTypes = {
+    containerStyle: PropTypes.object.isRequired,
+    filterGroup: PropTypes.object.isRequired,
+    selection: PropTypes.array.isRequired,
+    onUserInput: PropTypes.func.isRequired
+  };
+
+  handleChange = (value) => {
+    this.props.onUserInput(
+      this.props.filterGroup.name,
+      value,
+      true
+    );
+  };
+
+  itemStyle(index) {
+    const value = this.props.selection[0];
+    const selectedIndex = this.props.filterGroup.entries.findIndex(item => (item.name === value));
+
+    // When we have two unselected items next to each other, we want to draw a grey
+    // vertical divider between them, and that's done by rendering a border-left
+    // of the item on the right.
+
+    if (index === selectedIndex) {
```

Comment:
This seems complicated, but I don't know of a better way to do it. Consolidate both of the conditions that `return {}` ? I can see how it is more clear what's happening if they are separate. Doesn't need to be changed - just noting it took me a  minute to work through what was happening. Comments helped!